Tests/v3LoadShedderIntegration/v3LoadShedderIntegrationTests/v3LoadShedderIntegrationTests.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  v3LoadShedderIntegrationTests.swift
+//
+//  Created by Andrés Boedo on 3/29/23.
+
+@testable import Purchases
+import StoreKit
+import StoreKitTest
+@testable import v3LoadShedderIntegration
+import XCTest
+
+final class v3LoadShedderIntegrationTests: XCTestCase {
+    let apiKey = "API_KEY"
+
+    let skConfigFileName = "v3LoadShedderIntegrationTestsConfiguration"
+    let entitlementIdentifier = "premium"
+
+    var testSession: SKTestSession!
+
+    override func setUpWithError() throws {
+        Purchases.logLevel = .debug
+        let userDefaultsSuite = "v3LoadShedderIntegrationTests"
+        let userDefaults = UserDefaults(suiteName: userDefaultsSuite)!
+        userDefaults.removePersistentDomain(forName: userDefaultsSuite)
+        try configureTestSession()
+
+        Purchases.configure(withAPIKey: self.apiKey,
+                            appUserID: nil,
+                            observerMode: false,
+                            userDefaults: userDefaults)
+        clearReceiptIfExists()
+    }
+
+    func configureTestSession() throws {
+        assert(self.testSession == nil, "Attempted to configure session multiple times")
+
+        self.testSession = try SKTestSession(configurationFileNamed: skConfigFileName)
+        self.testSession.resetToDefaultState()
+        self.testSession.disableDialogs = true
+        self.testSession.clearTransactions()
+    }
+
+    func clearReceiptIfExists() {
+        let manager = FileManager.default
+
+        guard let url = Bundle.main.appStoreReceiptURL, manager.fileExists(atPath: url.path) else { return }
+
+        do {
+            print("Removing receipt from url: \(url)")
+            try manager.removeItem(at: url)
+        } catch {
+            print("Error attempting to remove receipt URL '\(url)': \(error)")
+        }
+    }
+
+    override func tearDownWithError() throws {
+        // unused for now
+    }
+
+    func testGetOfferings() async throws {
+        let offerings = try await Purchases.shared.offerings()
+        let offering = try XCTUnwrap(offerings.current)
+        XCTAssert(offering.availablePackages.count > 0)
+        let package = try XCTUnwrap(offering.availablePackages[0])
+        XCTAssert(package.product.productIdentifier == "com.revenuecat.loadShedder.monthly")
+        XCTAssert(offering.identifier == "default")
+    }
+
+    func testPurchasePackage() async throws {
```

Comment:
Should we have a test for `GetCustomerInfo` after a successful purchase? I guess the load shedder should also return the entitlement info right?

Reply:
the load shedder always returns 5xx for getCustomerInfo, it doesn't have enough context on any given subscriber to provide accurate data. So having a test for it won't really add anything. 
The idea is that if customerInfo is already in cache, the SDK will return the cached version anyway, so the system should work for all but fresh installs. 
And for the case of fresh installs, well, we're building Offline Entitlements

Reply:
Ah right, makes sense! 👍